lib/protocol/ConnectionManager.js

```diff
@@ -13,52 +13,49 @@
 // language governing permissions and limitations under the License.
 'use strict';
 
-var State = require('./ConnectionManagerState');
 var util = require('../util');
 var debug = util.debuglog('hdb_conn');
 
 module.exports = ConnectionManager;
 
 function ConnectionManager(settings) {
-  this.state = new State(settings);
+  this._settings = settings;
 }
 
-ConnectionManager.prototype.updateState = function updateState(newOptions) {
-  this.state.update(newOptions);
-};
-
-ConnectionManager.prototype.getConnectOptions = function getConnectOptions() {
-  return this.state.options.connect;
+ConnectionManager.prototype.updateConnectOptions = function updateConnectOptions(settings) {
+  return util.extend(this._settings, settings);
 };
 
 ConnectionManager.prototype.openConnection = function openConnection(conn, cb) {
-  var dbHosts = this.state.dbHosts;
+  var s = this._settings;
+  var dbHosts = (Array.isArray(s.hosts) && s.hosts.slice()) || [{'host': s.host, 'port': s.port}]
```

Comment:
- missing semicolon at the end of the line
- no need for the single quotes around `host` and `port`